File: src/engine/commands/equipmentcommand.cpp
#include "equipmentcommand.h"


#define super Command

EquipmentCommand::EquipmentCommand(QObject *parent) :
    super(parent) {

    setDescription("View any equipment you are currently wielding.\n"
                   "\n"
                   "Example: equipment");
}

EquipmentCommand::~EquipmentCommand() {
}

void EquipmentCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    GameObjectPtrList wieldedEquipment;
    if (!player->weapon().isNull()) {
        wieldedEquipment << player->weapon();
    }
    if (!player->secondaryWeapon().isNull()) {
        wieldedEquipment << player->secondaryWeapon();
    }
    if (!player->shield().isNull()) {
        wieldedEquipment << player->shield();
    }

    if (wieldedEquipment.isEmpty()) {
        send("You're not wielding anything.");
    } else {
        send(QString("You're wielding %1.").arg(wieldedEquipment.joinFancy()));
    }
}

File: data/commands/equipmentcommand.js
function EquipmentCommand() {
    Command.call(this);

    this.setDescription("View any equipment you are currently wielding.\n" +
                        "\n" +
                        "Example: equipment");
}

EquipmentCommand.prototype = new Command();
EquipmentCommand.prototype.constructor = EquipmentCommand;

EquipmentCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    var wieldedEquipment = [];
    if (player.weapon !== null) {
        wieldedEquipment.append(player.weapon);
    }
    if (player.secondaryWeapon !== null) {
        wieldedEquipment.append(player.secondaryWeapon);
    }
    if (player.shield !== null) {
        wieldedEquipment.append(player.shield);
    }

    if (wieldedEquipment.isEmpty()) {
        player.send("You're not wielding anything.");
    } else {
        player.send("You're wielding %1.".arg(wieldedEquipment.joinFancy()));
    }
}

CommandRegistry.registerCommand("equipment", new EquipmentCommand());
